### DISCUSSION TITLE
Does 'use client' in the root layout make whole routes client components?

### DISCUSSION THREAD

**Discussion by AxelReid** [vercel/next.js#44648] [Help] [answered]
I asked a question in [stackoverflow](https://stackoverflow.com/questions/74992326/does-use-client-in-next-js-13-root-layout-make-whole-routes-client-component).

Generally, what i'm confused that does treating parent as client component(cc) make the children cc too?

For example, `'use client'` in  root layout.
This can happen when using client libraries that require its provider to wrap the root

Like shown in the [docs](https://beta.nextjs.org/docs/rendering/server-and-client-components#rendering-third-party-context-providers-in-server-components). Creating a client component and wrap the root with that component. How is it different from directly specifing the `'use client'` in the root. And still the question, does it also make the all the children as cc?

Another example:
I have two components, `Projects.tsx ('use client' specified)`, `Project.tsx ('use client' not specified)` where the 'project' is imported into 'projects'. This is what i did yesterday, i think `Project` is imported as client even though no `use client` specified. This is one of the reasons i'm suspecting that the direct 'use client' or with extra client wrap in the root making the whole `app` cc.

**Comment by AxelReid** [comment]
Answer: [https://stackoverflow.com/a/75480826/16945230](https://stackoverflow.com/a/75480826/16945230)

**Reply by Amit-K-007** [reply]
I have explained this with eample. 
[https://stackoverflow.com/a/79367309/22304939](https://stackoverflow.com/a/79367309/22304939)
Do checkout and upvote if it helped :)

**Comment by AxelReid** [comment, marked as answer]
https://stackoverflow.com/a/75480826/16945230
